common/djangoapps/student/views.py

```diff
@@ -1230,30 +1232,46 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
 
         # if we didn't find this username earlier, the account for this email
         # doesn't exist, and doesn't have a corresponding password
+        msg = None
         if username != "":
             if settings.FEATURES['SQUELCH_PII_IN_LOGS']:
                 loggable_id = user_found_by_email_lookup.id if user_found_by_email_lookup else "<unknown>"
                 AUDIT_LOG.warning(u"Login failed - password for user.id: {0} is invalid".format(loggable_id))
             else:
                 AUDIT_LOG.warning(u"Login failed - password for {0} is invalid".format(email))
 
-        # Labster changes
-        # If there was login error, suggest student to login to region server
-        regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
-        # List of country codes: https://dev.maxmind.com/geoip/legacy/codes/iso3166/
-        current_region = request.session.get('country_code')
-        if current_region in regions.keys():
-            region_info = regions[current_region]
-            msg = _(
-                'If you are a student in {region_name}, your university may now be using the {region_code} version of '
-                'Labster. Please try to log into <a href="{region_login_url}">{region_name}</a> region.'
-            ).format(
-                region_name=region_info['name'],
-                region_code=region_info['region_code'],
-                region_login_url=region_info['login_url'],
-            )
+        # Start: Added by Labster
         else:
-            msg = _('Email or password is incorrect.')
+            # If there was login error, check whether user with the given email exists in one of the regions. If yes,
+            # suggest student to to login to the appropriate region server.
+            regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
+            region = get_user_region(request, regions, email, enrollment=False)
+            if region:
+                msg = _(
+                    'It appears that your account is located on our {region_code} server. '
+                    'Please log in <a href="{login_url}">here</a> instead.'
+                ).format(
+                    region_code=exc.region['region_code'],
+                    login_url=exc.region['login_url'],
+                )
+            else:
+                current_region = request.session.get('country_code')
+                region = regions.get(current_region)
+                # If ENABLE_REGION_IPADDR_WARNING is enabled, suggest student to login to
+                # the appropriate region server based on IP address of user.
+                if settings.LABSTER_FEATURES.get('ENABLE_REGION_IPADDR_WARNING') and region:
```

Comment:
please use `elif` as below

```py
if region:
  ...
elif settings.LABSTER_FEATURES.get('ENABLE_REGION_IPADDR_WARNING'):
  ...
else:
   msg = _("Account doesn't exist.")

Reply:
what if the user from IP that not listed in our region ex: `ID` ? does it will show `email or password incorrect` msg instead of 'Account doesn't exist.' ?

Reply:
it will say `Email or password is incorrect.`. But if you want `Account doesn't exist.` to be shown, use the following code
```py
if region:
  ...
elif settings.LABSTER_FEATURES.get('ENABLE_REGION_IPADDR_WARNING'):
  ...
msg = msg or _("Account doesn't exist.")
```

Reply:
updated :)